base/reflection.jl

```diff
@@ -153,6 +153,8 @@ methods(f::ANY,t::ANY) = methods(call, tt_cons(isa(f,Type) ? Type{f} : typeof(f)
 function _methods(f::ANY,t::ANY,lim)
     if isa(t,Type)
         _methods(f, Any[t.parameters...], length(t.parameters), lim, [])
+    elseif isa(t,TypeVar)
```

Comment:
When does this case arise? `_methods` generally expects either a tuple type or an iterable. This would be especially confusing if `t.ub` were a tuple type.